spec/unit/facter/udevadm_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'udevadm fact', type: :fact do
```

Comment:
Structure wise I I think what you can do is:
```ruby
describe 'udevadm fact', type: :fact do
  context 'on Linux' do
    before { allow(Facter.fact(:kernel)).to receive(:value).and_return('Linux') }

    context 'when udevadm is present' do
      # ...
    end

    context 'when udevadm is absent' do
      # ...
    end
  end

  context 'on Windows' do
    # ...
  end
end

Reply:
I honestly didn't do that on purpose as I'm not a fan on tests been deeply nested/indented but I will DRY it up.

Reply:
The spec has been DRY'd a bit.